.github/workflows/ci-2.x.yml

```diff
@@ -95,23 +120,94 @@ jobs:
         if: matrix.symfony == '5.*'
         run: SYMFONY_REQUIRE="^5.4" composer update --no-progress --ansi --prefer-stable
 
-      - name: Update project dependencies (6.*)
-        if: matrix.symfony == '6.*'
-        run: SYMFONY_REQUIRE="^6.0" composer update --no-progress --ansi --prefer-stable
-
       - name: Install PHPStan
         run: composer install --no-progress --ansi --working-dir=tools/phpstan
 
-      - name: Install PHP-CS-Fixer
-        run: composer install --no-progress --ansi --working-dir=tools/php-cs-fixer
+      - name: Run PHPStan
+        run: composer phpstan
 
-      - name: Validate composer
-        run: composer validate --strict --no-check-lock
+      - name: Run tests
+        run: vendor/bin/phpunit -c phpunit.${{ matrix.db }}.xml
 
-      - name: Run PHP-CS-Fixer
+  tests_sf_6:
+    name: Tests PHP ${{ matrix.php }}, Symfony ${{ matrix.symfony }}, ${{ matrix.db }}
+    runs-on: ubuntu-latest
```

Comment:
you could add a dependency with previous job to prevent this one to trigger if parent job has failed

```yaml
needs: [ code_style_composer_checks ]
```